File: framework/languages/java/e2e/RegistrationCodec/Client/src/main/java/systems/zlink/e2e/registrationcodec/client/Scenarios/AutoRegistrationScenario.java
package systems.zlink.e2e.registrationcodec.client.Scenarios;

import systems.zlink.e2e.registrationcodec.client.Support.ScenarioAssert;
import systems.zlink.e2e.registrationcodec.client.Support.ScenarioContext;
import systems.zlink.e2e.registrationcodec.shared.Contracts;

public final class AutoRegistrationScenario {
    private AutoRegistrationScenario() {
    }

    public static void run(ScenarioContext context) {
        Contracts.EchoRes auto = context.server().post("/registration/auto").submit(Contracts.EchoRes.class).toCompletableFuture().join().body();
        ScenarioAssert.ensure("echo:auto-request".equals(auto.value()) && "auto".equals(auto.handler()),
            "RC-A1 request mismatch");
        ScenarioAssert.waitForEvidence(context.evidence(), "Send", "EchoAuto", "auto-send");
        System.out.println("scenario RC-A1 passed");
    }
}

File: framework/languages/java/e2e-kotlin/RegistrationCodec/Client/src/main/kotlin/systems/zlink/e2e/kotlin/registrationcodec/client/Scenarios/AutoRegistrationScenario.kt
package systems.zlink.e2e.kotlin.registrationcodec.client.scenarios

import systems.zlink.e2e.kotlin.registrationcodec.EchoAutoRes
import systems.zlink.e2e.kotlin.registrationcodec.client.support.ScenarioAssert
import systems.zlink.e2e.kotlin.registrationcodec.client.support.ScenarioHttpClient

class AutoRegistrationScenario(
    private val server: ScenarioHttpClient,
    private val assert: ScenarioAssert,
) {
    fun run() {
        val auto = server.post<EchoAutoRes>("/registration/auto")
        assert.that(auto.value == "echo:auto-request" && auto.handler == "auto", "RC-A1 request mismatch")
        assert.that(
            server.waitForEvidence("Send", "EchoAutoMsg", "auto-send")
                .any { it.marker == "Send" && it.packetName == "EchoAutoMsg" && it.value == "auto-send" },
            "RC-A1 send evidence missing",
        )
        println("scenario RC-A1 passed")
    }
}
